### ISSUE TITLE
Unused private field false positive when called on other objects

### ISSUE THREAD

**Issue by jdufresne** [phpmd/phpmd#230]
Observe the following test case:

```
<?php

class foo {
    private $bar;

    public static function create() {
        $foo = new foo();
        $foo->bar = 'baz';
    }
}
```

```
$ phpmd test.php text unusedcode

/home/jon/test.php:4	Avoid unused private fields such as '$bar'.
```

The member `$bar` is not unused as it is used in the `foo::create()` method. I think the confusion is due to the fact that it is called on an object other than `$this`. This is legal PHP as stated in the documentation: <http://php.net/manual/en/language.oop5.visibility.php#language.oop5.visibility-other-objects>

> Objects of the same type will have access to each others private and protected members even though they are not the same instances. This is because the implementation specific details are already known when inside those objects. 

```
$ phpmd --version
PHPMD 1.5.0 by Manuel Pichler
```

**Comment by mikebronner** [comment]
I'm seeing this issue as well in version 2.4.3 in the following instance:

```php
class Test
{
    private $test;

    public function test()
    {
        $this->test = 5;
    }
}
```

**Comment by ravage84** [comment]
Duplicate of or at least related to #252
